engine/uState.ml

```diff
@@ -148,7 +148,25 @@ let of_binders b =
   in
   { ctx with uctx_names = b, rmap }
 
-let universe_binders ctx = fst ctx.uctx_names
+let invent_name (named,cnt) u =
+  let rec aux i =
```

Comment:
This looks horribly inefficient. Please use a cleverer algorithm, otherwise people with millions of universes will cry.

Reply:
The is the clever version, how would you do better?

Reply:
Yeah, you're right, it's probably efficient enough... I'd be curious about a benchmark though, just in case.